pkg/operator/controllers/checker/serviceprincipalchecker.go

```diff
@@ -69,29 +72,35 @@ func (r *ServicePrincipalChecker) Check(ctx context.Context) error {
 		return err
 	}
 
-	_, err = aad.GetToken(ctx, r.log, string(azCred.ClientID), string(azCred.ClientSecret), string(azCred.TenantID), azEnv.ActiveDirectoryEndpoint, azEnv.ResourceManagerEndpoint)
+	_, err = r.tokenClient.GetToken(ctx, r.log, string(azCred.ClientID), string(azCred.ClientSecret), string(azCred.TenantID), azEnv.ActiveDirectoryEndpoint, azEnv.ResourceManagerEndpoint)
 	if err != nil {
 		updateFailedCondition(cond, err)
 	}
 
-	spDynamic, err := dynamic.NewServicePrincipalValidator(r.log, &azEnv, dynamic.AuthorizerClusterServicePrincipal)
+	spDynamic, err := dynamic.NewServicePrincipalValidator(r.log, &azEnv, dynamic.AuthorizerClusterServicePrincipal, aad.NewTokenClient())
 	if err != nil {
 		return err
 	}
 
-	err = spDynamic.ValidateServicePrincipal(ctx, string(azCred.ClientID), string(azCred.ClientSecret), string(azCred.TenantID))
+	if r.validateServicePrincipal != nil {
+		err = r.validateServicePrincipal.ValidateServicePrincipal(ctx, string(azCred.ClientID), string(azCred.ClientSecret), string(azCred.TenantID))
+	} else {
+		err = spDynamic.ValidateServicePrincipal(ctx, string(azCred.ClientID), string(azCred.ClientSecret), string(azCred.TenantID))
+	}
+
 	if err != nil {
 		updateFailedCondition(cond, err)
 	}
 
 	return conditions.SetCondition(ctx, r.arocli, cond, r.role)
 }
 
-func updateFailedCondition(cond *operatorv1.OperatorCondition, err error) {
+func updateFailedCondition(cond *operatorv1.OperatorCondition, err error) *operatorv1.OperatorCondition {
 	cond.Status = operatorv1.ConditionFalse
 	if tErr, ok := err.(*api.CloudError); ok {
 		cond.Message = tErr.Message
 	} else {
 		cond.Message = err.Error()
 	}
+	return cond
```

Comment:
Actually this is redundant. we are already working with a pointer so the change will be reflected onto the struct. 
```suggestion
```

Reply:
Thanks for pointing that out. I've removed this and squashed my commits.